server/api/api/endpoints/functions.py

```diff
@@ -790,58 +787,7 @@ def _deploy_nuclio_runtime(
             )
         else:
             model_monitoring_access_key = None
-        if serving_to_monitor:
-            # TODO : delete when batch is deprecated.
-            _deploy_serving_monitoring(
-                auth_info,
-                db_session,
-                fn,
-                model_monitoring_access_key,
-            )
-        if monitoring_application:
-            monitoring_deploy = (
-                server.api.crud.model_monitoring.deployment.MonitoringDeployment(
-                    project=fn.metadata.project,
-                    auth_info=auth_info,
-                    db_session=db_session,
-                    model_monitoring_access_key=model_monitoring_access_key,
-                )
-            )
-            fn = monitoring_deploy._apply_and_create_stream_trigger(
-                function=fn,
-                function_name=fn.metadata.name,
-            )
-    server.api.crud.runtimes.nuclio.function.deploy_nuclio_function(
-        fn,
-        auth_info=auth_info,
-        client_version=client_version,
-        client_python_version=client_python_version,
-        builder_env=builder_env,
-    )
-    return fn
-
 
-def _deploy_serving_monitoring(
-    auth_info,
-    db_session,
-    fn,
-    model_monitoring_access_key,
-):
-    # TODO : delete when batch is deprecated.
-    try:
-        # Handle model monitoring
-        logger.info("Tracking enabled, initializing model monitoring")
-
-        if fn.spec.tracking_policy:
-            # Convert to `TrackingPolicy` object as `fn.spec.tracking_policy` is provided as a dict
-            fn.spec.tracking_policy = TrackingPolicy.from_dict(fn.spec.tracking_policy)
-        else:
-            # Initialize tracking policy with default values
-            fn.spec.tracking_policy = TrackingPolicy()
-
-        if not mlrun.mlconf.is_ce_mode():
-            _init_serving_function_stream_args(fn=fn)
```

Comment:
As far as I know In case of serving_to_monitor & v3io usage,  you have to run this line, because the serving needs the stream_args which this function initializing in order to send the events to the stream pod. Are you sure all the test are passed without it?

Reply:
Brought it back in a3fbc0b.
I'll open issues to investigate this issue further, as we discussed orally.